### DISCUSSION TITLE
Help using the streamlink API.

### DISCUSSION THREAD

**Discussion by NeoFahrenheit** [streamlink/streamlink#4880] [Q&A] [answered]
Hi guys, I'm trying to use streamlink to download streams, like on Twitch.tv. My previous aproach was using subprocess to run streamlink through a terminal, but that [wans't successful](https://stackoverflow.com/questions/74002131/reading-the-countinous-output-of-a-streamlink-cli-with-subprocess-popen). I opened a discussion [here](https://github.com/streamlink/streamlink/discussions/4878) about it, but **[bastimeyer](https://github.com/bastimeyer)** said I would better using the streamlink API and the CLI download strings are outputed to stderr. I tried to read from stderr and got nothing. 

So what's left for me is the streamlink API. I have read the documentation, searched for code examples throughout the internet and nothing helped me to get a simple stream, write it's data to the disk and get information like the file size that has already been downloaded, transfer speed and time elapsed of the recording, etc.

So, can someone help me with that, please?
I need to write a stream to the disk, read some download status and to know when it's over. 

Thank you!

**Comment by gravyboat** [comment, marked as answer]
@bastimeyer Already provided all the help we can provide in the [response](https://github.com/streamlink/streamlink/discussions/4878#discussioncomment-3852723) you linked. Past that answer, the [API documentation](https://streamlink.github.io/api_guide.html), and the [API reference documentation](https://streamlink.github.io/api.html) there's nothing else we can do for you. We are a small team and we don't have time to provide any support when it comes to developing third party tools or building applications. Please don't create another discussion thread as there's nothing else we can help you with.

**Comment by bastimeyer** [comment]
> searched for code examples throughout the internet

Take a look at Streamlink's code, not some random stuff you've found via a search engine.
